app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,19 @@ def show
     @post = Post.find(params[:id])
     @comments = @post.comments
   end
+
+  def new
+    @post = Post.new
+    # @user=current_user
+  end
+
+  def create
+    @new_post = current_user.posts.create(title: params[:post][:title], text: params[:post][:text],
+                                          author_id: params[:user_id])
```

Comment:
- [ ] [OPTIONAL] Good job working on this, but it would be better if you used the strong parameters method. Here is a [link](`https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides`) to the lesson on strong parameters

```ruby
private 
    def post_params
      params.require(:post).permit(:title, :text)
    end

```